wfe/wfe_test.go

```diff
@@ -656,6 +666,34 @@ func TestDirectory(t *testing.T) {
 	test.AssertEquals(t, responseWriter.Header().Get("Content-Type"), "application/json")
 	test.AssertEquals(t, responseWriter.Code, http.StatusOK)
 	assertJSONEquals(t, responseWriter.Body.String(), `{"new-authz":"http://localhost:4300/acme/new-authz","new-cert":"http://localhost:4300/acme/new-cert","new-reg":"http://localhost:4300/acme/new-reg","revoke-cert":"http://localhost:4300/acme/revoke-cert"}`)
+
+	_ = features.Set(map[string]bool{"RandomDirectoryEntry": true})
+	origRead := rand.Reader
+	defer func() { rand.Reader = origRead }()
+	rand.Reader = &fakeReader{}
+
+	responseWriter.Body.Reset()
+	url, _ = url.Parse("/directory")
+	mux.ServeHTTP(responseWriter, &http.Request{
+		Method: "GET",
+		URL:    url,
+		Host:   "127.0.0.1:4300",
+	})
+	test.AssertEquals(t, responseWriter.Header().Get("Content-Type"), "application/json")
+	test.AssertEquals(t, responseWriter.Code, http.StatusOK)
+	assertJSONEquals(t, responseWriter.Body.String(), `{"AQEBAQEBAQE":"https://community.letsencrypt.org/t/adding-random-entries-to-the-directory/33417","key-change":"http://localhost:4300/acme/key-change","meta":{"terms-of-service":"http://example.invalid/terms"},"new-authz":"http://localhost:4300/acme/new-authz","new-cert":"http://localhost:4300/acme/new-cert","new-reg":"http://localhost:4300/acme/new-reg","revoke-cert":"http://localhost:4300/acme/revoke-cert"}`)
```

Comment:
Doing an assertJSONEquals here will make this test brittle with respect to new directory entries. Instead, please write a narrower test that parses the JSON into a map and checks that one of the values equals the explanation link. That will also allow you to avoid having to mock out rand.Reader.